### PR TITLE
rocAL - Python Unit test fix

### PR DESCRIPTION
**Pull request by swetha097**
The python unit test breaks at TOT. This PR fixes issues with Python unit test

### CODE REVIEW

**Review by rrawther — APPROVED** [review]
@paveltc to make sure tests are passing

**Comment by paveltc** [comment]
@rrawther Tests ran successfully. 
@swetha097 How many tests are supposed to run for the CPU and how many for the GPU?

File: rocAL/rocAL_pybind/amd/rocal/pipeline.py [hunk 1]
```diff
@@ -215,18 +215,18 @@ def copyToExternalTensorNCHW(self, array,  multiplier, offset, reverse_channels,
...
             else: #torch tensor
                 return b.getOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses, 0)
-        if device=="gpu":
+        else:
             if (isinstance(array,cp.ndarray)):
-                b.getOneHotEncodedLabels(self._handle, array.data.ptr, self._numOfClasses, 1)
+                b.getCupyOneHotEncodedLabels(self._handle, array.data.ptr, self._numOfClasses, 1)
             else: #torch tensor
-                return b.getOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses, 1)
+                return b.getCupyOneHotEncodedLabels(self._handle, ctypes.c_void_p(array.data_ptr()), self._numOfClasses, 1)
```
Comment (LakshmiKumar23) [inline rocAL/rocAL_pybind/amd/rocal/pipeline.py:229]: @swetha097 Why are we changing the torch part to use cupy as well? I thought only tensorflow and bare metal GPU uses the CuPy flow and torch uses it's own tensor. Can you check on this?

**Review by LakshmiKumar23 — CHANGES_REQUESTED** [review]

File: rocAL/rocAL_pybind/amd/rocal/pipeline.py (hunk 1 above)
Comment (swetha097) [reply rocAL/rocAL_pybind/amd/rocal/pipeline.py]: Changed it back to using getOneHotEncodedLabels API for the torch GPU tensors.

**Review by LakshmiKumar23 — APPROVED** [review]
@kiritigowda can be merged

**Comment by paveltc** [comment]
@kiritigowda Fine to merge this.

**Comment by paveltc** [comment]
> > @swetha097 I am only seeing one test run the GPU log versus the 31 in the CPU log. @LakshmiKumar23 said she saw the same thing. Can this be fixed? Thanks, Pavel
> 
> @swetha097 this is because the shell script does not accept -b gpu when you pass with ./PYTHON_UNITTEST_TEST.sh

Is it possible to create a fix for this? Is anyone working on this?

**Comment by swetha097** [comment]
> > > @swetha097 I am only seeing one test run the GPU log versus the 31 in the CPU log. @LakshmiKumar23 said she saw the same thing. Can this be fixed? Thanks, Pavel
> > 
> > 
> > @swetha097 this is because the shell script does not accept -b gpu when you pass with ./PYTHON_UNITTEST_TEST.sh
> 
> Is it possible to create a fix for this? Is anyone working on this?

@paveltc @LakshmiKumar23  I am not seeing this issue in the docker I am testing on. Can you please provide me more details on which machine and docker this is tested on

**Pull request merged**